lib/workers/repository/update/branch/auto-replace.ts

```diff
@@ -63,35 +55,56 @@ export async function confirmIfDepUpdated(
     return false;
   }
 
-  if (newValue && newUpgrade.currentValue !== newValue) {
+  if (upgrade.newName && upgrade.newName !== newUpgrade.depName) {
+    logger.debug(
+      {
+        manager,
+        packageFile,
+        currentDepName: upgrade.depName,
+        newDepName: newUpgrade.depName,
+      },
+      'depName is not updated'
+    );
+    return false;
+  }
+
+  if (upgrade.newValue && upgrade.newValue !== newUpgrade.currentValue) {
     logger.debug(
       {
         manager,
         packageFile,
-        expectedValue: newValue,
+        expectedValue: upgrade.newValue,
         foundValue: newUpgrade.currentValue,
       },
       'Value is not updated'
     );
     return false;
   }
 
-  if (!newDigest) {
-    return true;
-  }
-  if (newUpgrade.currentDigest === newDigest) {
-    return true;
-  }
-  if (!currentDigest) {
-    if (!pinDigests) {
-      return true;
-    } else if (newDigest) {
-      return true;
+  if (upgrade.newDigest) {
+    if (
+      (upgrade.pinDigests &&
+        !upgrade.currentDigest &&
+        upgrade.updateType === 'pinDigest') ||
+      upgrade.currentDigest
+    ) {
+      if (upgrade.newDigest !== newUpgrade.currentDigest) {
```

Comment:
can we combine to one condition? i only see 3 nested if and no else.

Reply:
I can do if we feel that's more legible.  I left them split out initially as I found it much easier to comprehend to different scenarios being tested for

Reply:
I'd like to have less nested code, which is easier  to read for me

Reply:
will update the PR shortly